### ISSUE TITLE
Core component: Select field

### ISSUE THREAD

**Issue by chmanie** [JoinColony/colonyDapp#184]
### Description

This should be in a good state, just to be moved to flow and formik. Make sure, the a11y doesn't get killed.

### Related files in colonyBeta

* https://github.com/JoinColony/colonyBeta/tree/develop/src/client/modules/core/components/Fields/Select

### Screencaps

![image](https://user-images.githubusercontent.com/2174084/43611319-fafd2784-96a8-11e8-9b9f-a02cbb79734b.png)
![image](https://user-images.githubusercontent.com/2174084/43611324-01420a06-96a9-11e8-93cd-5b0de140f192.png)

### Examples in figma

https://www.figma.com/file/g5oQtPcDUAhvy1pPHFG7VFrx/High-Level---Colony-Contribute?node-id=68%3A0

https://www.figma.com/file/ajCQLCWUGh6gkELhfnptctQZ/Dashboard?node-id=546%3A225
